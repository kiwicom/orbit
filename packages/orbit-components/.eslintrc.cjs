const path = require("path");
const orbitPreset = require("@kiwicom/orbit-tailwind-preset");

module.exports = {
  extends: ["../../.eslintrc.cjs", "plugin:storybook/recommended"],
  rules: {
    "import/no-extraneous-dependencies": [
      "error",
      {
        packageDir: __dirname,
      },
    ],
  },
  env: {
    browser: true,
  },
  settings: {
    tailwindcss: {
      config: orbitPreset(),
    },
  },
  overrides: [
    {
      files: ["cypress/**/*.ts?(x)", "config/**/*.mts"],
      rules: {
        "import/no-extraneous-dependencies": "off",
      },
    },
    {
      files: ["{src,es,lib}/**/*.js?(x)", "*.js?(x).flow"],
      rules: {
        "flowtype/require-valid-file-annotation": ["error", "always"],
      },
    },
    // TODO: temporary disabled because of the issue: https://github.com/typescript-eslint/typescript-eslint/issues/4608
    {
      files: ["src/**/*.ts?(x)"],
      rules: {
        "@typescript-eslint/consistent-type-imports": "warn",
      },
    },
    {
      files: ["*.js?(x)", "*.js?(x).flow"],
      parser: "@babel/eslint-parser",
      extends: ["plugin:flowtype/recommended", "prettier"],
      parserOptions: {
        ecmaVersion: 2020,
        sourceType: "module",
        requireConfigFile: false,
      },
      rules: {
        "flowtype/require-exact-type": "error",
        "react/default-props-match-prop-types": "off", // Conflict between Flow and ESLint
      },
    },
    {
      files: "*.stories.*",
      rules: {
        "orbit-components/unique-id": "off",
        "react/no-unstable-nested-components": "off",
        // TODO: remove after migration to a new actions
        "import/no-extraneous-dependencies": "off",
      },
    },
    {
      files: ["src/utils/**/*"],
      rules: {
        "@typescript-eslint/prefer-readonly-parameter-types": "off",
      },
    },
    {
      files: ["scripts/*"],
      rules: {
        "no-console": "off",
        "no-restricted-syntax": "off",
      },
    },
    {
      files: ["src/**/*.test.ts?(x)", "src/test-utils.tsx"],
      rules: {
        "orbit-components/unique-id": "off",
        "import/no-extraneous-dependencies": [
          "error",
          {
            packageDir: [__dirname, path.resolve(__dirname, "../../")],
          },
        ],
      },
    },
    {
      files: ["config/**", ".storybook/**"],
      rules: {
        "@typescript-eslint/prefer-readonly-parameter-types": "off",
        "@typescript-eslint/explicit-function-return-type": "off",
        "@typescript-eslint/explicit-module-boundary-types": "off",
        "no-restricted-syntax": "off",
        "guard-for-in": "off",
        "no-console": "off",
        "import/extensions": "off",
        "import/no-extraneous-dependencies": [
          "error",
          {
            packageDir: [__dirname, path.resolve(__dirname, "../../")],
          },
        ],
      },
    },
  ],
};
