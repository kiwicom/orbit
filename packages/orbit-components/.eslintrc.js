module.exports = {
  overrides: [
    {
      files: "cypress/**",
      rules: {
        "@typescript-eslint/prefer-readonly-parameter-types": "off",
        "@typescript-eslint/explicit-module-boundary-types": "off",
        "@typescript-eslint/no-unused-expressions": "off",
      },
    },
  ],
};
