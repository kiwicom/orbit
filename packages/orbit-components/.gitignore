lib
umd
es
.out
.cache
src/data/airportIllustrations.json
src/data/illustrations.json
src/data/icons.json
src/Illustration/index.js.flow
src/AirportIllustration/index.js.flow
src/ServiceLogo/index.js.flow
src/FeatureIcon/index.js.flow
src/Illustration/index.d.ts
src/AirportIllustration/index.d.ts
src/ServiceLogo/index.d.ts
src/FeatureIcon/index.d.ts
orbit-icons-font
orbit-email-icons
orbit-icons-font.zip
orbit-svgs.zip
