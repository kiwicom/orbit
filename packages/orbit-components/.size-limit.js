/* eslint-disable no-restricted-syntax */
const fs = require("fs");

const { peerDependencies } = require("./package.json");

const sortBy = key => {
  return (a, b) => {
    if (a[key] > b[key]) return 1;
    if (b[key] > a[key]) return -1;
    return 0;
  };
};

const exportMatches = fs
  .readFileSync(`${__dirname}/src/index.ts`)
  .toString()
  .matchAll(
    /export (\{ (default as )?(?<importedModules>.*) \}|\* as (?<importNamespace>.*)) from "(?<modulePath>.*)"/g,
  );

const entries = [];

for (const {
  groups: { importedModules, importNamespace, modulePath },
} of exportMatches) {
  const name = modulePath.startsWith(".")
    ? modulePath.replace(/^\.\//, "").split("/").slice(-1)[0]
    : modulePath;
  entries.push({
    name,
    path: `${__dirname}/es/index.js`,
    import: `{ ${importedModules || importNamespace} }`,
  });
}

module.exports = [
  {
    name: "esm",
    path: `${__dirname}/es/size-measurer.js`,
    import: "{ Orbit }",
    limit: "190 kB",
  },
  ...entries.sort(sortBy("name")),
].map(entry => ({
  ...entry,
  ignore: Object.keys(peerDependencies),
  modifyWebpackConfig(config) {
    // eslint-disable-next-line no-param-reassign
    config.resolve = {
      alias: {
        // resolve to .js rather than .mjs to avoid webpack failing to resolve modules
        "@adeira/js": require.resolve("@adeira/js/src/index.js"),
      },
    };
    return config;
  },
}));
