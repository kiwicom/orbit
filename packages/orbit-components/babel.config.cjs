module.exports = {
  presets: [
    [require.resolve("@babel/preset-env"), { bugfixes: true, loose: true }],
    [require.resolve("@babel/preset-react"), { runtime: "classic" }],
    [require.resolve("@babel/preset-typescript"), { isTSX: true, allExtensions: true }],
  ],
  plugins: [
    require.resolve("babel-plugin-styled-components"),
    [
      require.resolve("@babel/plugin-transform-runtime"),
      { version: require("@babel/runtime/package.json").version },
    ],
  ],
  env: {
    esm: {
      presets: [
        [require.resolve("@babel/preset-env"), { modules: false, bugfixes: true, loose: true }],
      ],
      plugins: [
        require.resolve("babel-plugin-styled-components"),
        [
          require.resolve("@babel/plugin-transform-runtime"),
          { useESModules: true, version: require("@babel/runtime/package.json").version },
        ],
      ],
    },
    test: {
      plugins: [
        require.resolve("babel-plugin-styled-components"),
        require.resolve("babel-plugin-require-context-hook"),
      ],
    },
  },
};
