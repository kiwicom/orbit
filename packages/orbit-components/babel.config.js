module.exports = {
  presets: [
    [require.resolve("@babel/preset-env"), { bugfixes: true, loose: true }],
    [require.resolve("@babel/preset-react"), { runtime: "classic" }],
    [require.resolve("@babel/preset-typescript"), { isTSX: true, allExtensions: true }],
  ],
  plugins: [
    require.resolve("babel-plugin-styled-components"),
    [require.resolve("@babel/plugin-proposal-class-properties"), { loose: true }],
    [require.resolve("@babel/plugin-proposal-private-methods"), { loose: true }],
    require.resolve("@babel/plugin-proposal-object-rest-spread"),
    require.resolve("@babel/plugin-proposal-nullish-coalescing-operator"),
    require.resolve("@adeira/babel-preset-adeira/src/adeira-js-warning"),
    require.resolve("@adeira/babel-preset-adeira/src/adeira-js-invariant"),
    [
      require.resolve("@babel/plugin-transform-runtime"),
      { version: require("@babel/runtime/package.json").version },
    ],
  ],
  env: {
    esm: {
      presets: [
        [require.resolve("@babel/preset-env"), { modules: false, bugfixes: true, loose: true }],
      ],
      plugins: [
        require.resolve("babel-plugin-styled-components"),
        [
          require.resolve("@babel/plugin-transform-runtime"),
          { useESModules: true, version: require("@babel/runtime/package.json").version },
        ],
      ],
    },
    test: {
      plugins: [
        require.resolve("babel-plugin-styled-components"),
        require.resolve("babel-plugin-require-context-hook"),
      ],
    },
  },
};
