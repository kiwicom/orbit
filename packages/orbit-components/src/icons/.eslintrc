{
  "rules": {
    "flowtype/require-valid-file-annotation": "off"
  }
}
