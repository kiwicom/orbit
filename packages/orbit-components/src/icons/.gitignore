/*.js
/*.jsx
/*.ts
/*.tsx
/*.d.ts
