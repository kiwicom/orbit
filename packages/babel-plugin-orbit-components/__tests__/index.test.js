// @flow

const os = require("os");
const { transform: babelTransform } = require("@babel/core");

const plugin = require("..");

expect.addSnapshotSerializer({
  print(serializerValue) {
    return Object.keys(serializerValue)
      .map(key => `~~~~~~~~~~ ${key.toUpperCase()} ~~~~~~~~~~\n${serializerValue[key]}`)
      .join(os.EOL);
  },
  test() {
    return true;
  },
});

function transform(input) {
  return babelTransform(input, {
    plugins: [plugin],
    root: __dirname,
    rootMode: "root",
  }).code;
}

// TODO: verify the generated output against the actual Orbit components somehow

test.each([
  [
    // This import path is unknown for this plugin is it's being ignored (returned as is).
    'import { Whatever } from "@kiwicom/orbit-components/unknown/import";',
  ],
  ['import { Alert } from "@kiwicom/orbit-components";'], // single
  ['import { Alert, Button } from "@kiwicom/orbit-components";'], // multi
  ['import Alert from "@kiwicom/orbit-components/lib/Alert";'], // from lib (stays as is)
  ['import { ModalSection } from "@kiwicom/orbit-components";'], // nested
  ['import { Alert, ModalSection } from "@kiwicom/orbit-components";'], // nested multi
  ['import { Passengers } from "@kiwicom/orbit-components/lib/icons";'], // icon
  ['import { LayoutColumn } from "@kiwicom/orbit-components";'],
  ['import { BreadcrumbsItem } from "@kiwicom/orbit-components";'],
  ['import { BadgePrimitive } from "@kiwicom/orbit-components";'],
  ['import { ListItem } from "@kiwicom/orbit-components";'],
  [
    // mixed
    `import { Alert, ModalSection, mediaQueries } from "@kiwicom/orbit-components";
import { Passengers, Invoice } from "@kiwicom/orbit-components/lib/icons";
import { Accommodation } from "@kiwicom/orbit-components/icons";`,
  ],

  // REPORTED ISSUES:
  ['import { mediaQueries } from "@kiwicom/orbit-components";'], // https://github.com/kiwicom/babel-plugin-orbit-components/pull/4
  ["import { Text, Stack, Grid } from '@kiwicom/orbit-components';"], // https://github.com/kiwicom/babel-plugin-orbit-components/issues/5
  ["import { Icon, Icons } from '@kiwicom/orbit-components'"], // https://github.com/kiwicom/babel-plugin-orbit-components/issues/6
])("transform imports as expected %#", rawInput => {
  const input = rawInput.trim();
  expect({
    input,
    output: transform(input),
  }).toMatchSnapshot();
});
