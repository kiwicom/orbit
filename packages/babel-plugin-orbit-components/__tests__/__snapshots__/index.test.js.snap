// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform imports as expected 0 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { Whatever } from "@kiwicom/orbit-components/unknown/import";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import { Whatever } from "@kiwicom/orbit-components/unknown/import";
`;

exports[`transform imports as expected 1 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { Alert } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import Alert from "@kiwicom/orbit-components/lib/Alert";
`;

exports[`transform imports as expected 2 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { Alert, Button } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import Button from "@kiwicom/orbit-components/lib/Button";
import Alert from "@kiwicom/orbit-components/lib/Alert";
`;

exports[`transform imports as expected 3 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import Alert from "@kiwicom/orbit-components/lib/Alert";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import Alert from "@kiwicom/orbit-components/lib/Alert";
`;

exports[`transform imports as expected 4 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { ModalSection } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import ModalSection from "@kiwicom/orbit-components/lib/Modal/ModalSection";
`;

exports[`transform imports as expected 5 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { Alert, ModalSection } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import ModalSection from "@kiwicom/orbit-components/lib/Modal/ModalSection";
import Alert from "@kiwicom/orbit-components/lib/Alert";
`;

exports[`transform imports as expected 6 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { Passengers } from "@kiwicom/orbit-components/lib/icons";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import Passengers from "@kiwicom/orbit-components/lib/icons/Passengers";
`;

exports[`transform imports as expected 7 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { LayoutColumn } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import LayoutColumn from "@kiwicom/orbit-components/lib/Layout/LayoutColumn";
`;

exports[`transform imports as expected 8 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { BreadcrumbsItem } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import BreadcrumbsItem from "@kiwicom/orbit-components/lib/Breadcrumbs/BreadcrumbsItem";
`;

exports[`transform imports as expected 9 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { BadgePrimitive } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import BadgePrimitive from "@kiwicom/orbit-components/lib/primitives/BadgePrimitive";
`;

exports[`transform imports as expected 10 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { ListItem } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import ListItem from "@kiwicom/orbit-components/lib/List/ListItem";
`;

exports[`transform imports as expected 11 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { Alert, ModalSection, mediaQueries } from "@kiwicom/orbit-components";
import { Passengers, Invoice } from "@kiwicom/orbit-components/lib/icons";
import { Accommodation } from "@kiwicom/orbit-components/icons";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import mediaQueries from "@kiwicom/orbit-components/lib/utils/mediaQuery";
import ModalSection from "@kiwicom/orbit-components/lib/Modal/ModalSection";
import Alert from "@kiwicom/orbit-components/lib/Alert";
import Invoice from "@kiwicom/orbit-components/lib/icons/Invoice";
import Passengers from "@kiwicom/orbit-components/lib/icons/Passengers";
import Accommodation from "@kiwicom/orbit-components/lib/icons/Accommodation";
`;

exports[`transform imports as expected 12 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { mediaQueries } from "@kiwicom/orbit-components";
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import mediaQueries from "@kiwicom/orbit-components/lib/utils/mediaQuery";
`;

exports[`transform imports as expected 13 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { Text, Stack, Grid } from '@kiwicom/orbit-components';
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import Grid from "@kiwicom/orbit-components/lib/utils/Grid";
import Stack from "@kiwicom/orbit-components/lib/Stack";
import Text from "@kiwicom/orbit-components/lib/Text";
`;

exports[`transform imports as expected 14 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
import { Icon, Icons } from '@kiwicom/orbit-components'
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
import Icons from "@kiwicom/orbit-components/lib/icons";
import Icon from "@kiwicom/orbit-components/lib/Icon";
`;
